spring-cloud-kubernetes-loadbalancer/src/main/java/org/springframework/cloud/kubernetes/loadbalancer/KubernetesServiceInstance.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.loadbalancer;
+
+import java.net.URI;
+import java.util.Map;
+
+import org.springframework.cloud.client.ServiceInstance;
+
+/**
+ * @author Piotr Minkowski
+ */
+public class KubernetesServiceInstance implements ServiceInstance {
```

Comment:
We already have a `ServiceInstance` implementation

Reply:
Yes, I know, but it is in disocvery module and uses `EndpointAddress` and `EndpointPort` objects, which are related to the Kubernetes endpoints not services

Reply:
Is there some kind of abstraction that can be done?  We certainly don't want 2 ServiceInstance implementations

Reply:
Ok, I changed implementation of `KubernetesServiceInstance` inside discovery module by replacing `EndpointAddress` and `EndpointPort` into `String host` and `int port`. Now I'm using that implementation in loadbalancer